lib/apiservers/engine/backends/system.go

```diff
@@ -381,3 +408,45 @@ func FetchVolumeStores(client *client.PortLayer) (string, error) {
 
 	return strings.Join(res.Payload.Stores, " "), nil
 }
+
+// RegistrySetContains checks if a URL is within the set of insecure registries
+func RegistrySetContains(registrySet []url.URL, remoteURL *url.URL) bool {
+	containedOK := false
+	remoteIP := net.ParseIP(remoteURL.Hostname())
+	remoteDomains, _ := net.LookupAddr(remoteURL.Hostname())
```

Comment:
You may not need to do this if none of the registry entries are domain names, wild-carded or otherwise.